react/components/form/ImageUploader/MediaCenter.js

```diff
@@ -0,0 +1,180 @@
+import React, { Fragment } from 'react'
+import { Button, Modal, Tabs, Tab, Spinner } from 'vtex.styleguide'
+import ImageIcon from '../../../images/ImageIcon'
+import Dropzone from './Dropzone'
+import Gallery from './Gallery'
+
+class MediaCenter extends React.Component {
+  state = {
+    currentTab: 1,
+    isLoading: false,
+    error: false,
+  }
+
+  handleClearImage = () => {
+    this.props.onChange(null)
+  }
+
+  handleGallerySelect = image => {
+    this.props.onChange(image)
+    this.props.closeModal()
+  }
+
+  handleImageDrop = async (acceptedFiles, rejectedFiles) => {
+    const { uploadFile } = this.props
+
+    if (acceptedFiles && acceptedFiles[0]) {
+      this.setState({ isLoading: true, error: false })
+
+      try {
+        const {
+          data: {
+            uploadFile: { fileUrl },
+          },
+        } = await uploadFile({
+          variables: { file: acceptedFiles[0] },
+        })
+
+        if (fileUrl) {
+          this.props.onChange(fileUrl)
+        }
+      } catch (e) {
+        this.setState({ error: true })
+        console.log('Error: ', e)
+      }
+
+      this.setState({ isLoading: false })
+    }
+
+    if (rejectedFiles && rejectedFiles[0]) {
+      this.setState({ error: true })
+      console.log(
+        'Error: one or more files are not valid and, therefore, have not been uploaded.'
+      )
+    }
+  }
+
+  handleTabChange = tabIndex => {
+    this.setState({
+      currentTab: tabIndex,
+      error: false,
+    })
+  }
+
+  render() {
+    const { value, disabled } = this.props
+    const { isLoading, error } = this.state
+
+    const backgroundImageStyle = {
+      backgroundImage: `url(${value})`,
+      backgroundPosition: 'center',
+    }
+
+    const isButtonDisabled = value == null
+
+    return (
```

Comment:
Isn't better to break in smaller components?

Reply:
👍